rpc/proto/mayastor.proto

```diff
@@ -368,3 +371,38 @@ message BdevUri {
 message CreateReply {
   string name = 1;
 }
+
+message BlockDevice {
+  message Partition {
+    string parent = 1;          // devname of parent device to which this partition belongs
+    uint32 number = 2;          // partition number
+    string name = 3;            // partition name
+    string scheme = 4;          // partition scheme: gpt, dos, ...
+    string typeid = 5;          // partition type identifier
+    string uuid = 6;            // UUID identifying partition
+  }
+  message Filesystem {
+    string fstype = 1;          // filesystem type: ext3, ntfs, ...
+    string label = 2;           // volume label
+    string uuid = 3;            // UUID identifying the volume (filesystem)
+    string mountpoint = 4;      // path where filesystem is currently mounted
+  }
+  string devname = 1;           // entry in /dev associated with device
+  string devtype = 2;           // currently "disk" or "partition"
+  uint64 devmajor = 3;          // major device number
+  uint64 devminor = 4;          // minor device number
```

Comment:
devmajor and devminor definitely don't need to be 64 bit ints. IIRC these are normally represented by a single byte in C. Would be good to keep the same size as the OS.

Reply:
Not quite a single byte.
Linux currently uses a 32-bit dev_t with 12 bits for major and 20 bits for minor.
Apparently glibc uses a 64-bit dev_t with 32 bits each for major and minor.
So probably best to change to u32 here.

Reply:
agree and thanks for looking up the right size 👍